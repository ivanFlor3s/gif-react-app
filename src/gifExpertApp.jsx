import React, {useState} from 'react'
import { AddCategory } from './components/AddCategory'
import { GifContainer } from './Container/GifContainer'

export const GifExpertApp = ({categoryList = []}) => {

    const [categorias, setCategorias] = useState(categoryList )

    // const handleAdd = ()=>{
    //     //setCategorias([...categorias, 'Boku no epicardo'])
    //     setCategorias((categs)=>[...categs,'Boku no Hero'])
    // }
    
    return (
        <>
            <h1>Gif Expert App</h1>
            <hr />
            <AddCategory addGifHandler={setCategorias}></AddCategory>
            <h3>Categorias</h3>
            
            {
                categorias.map( c => {
                   return <GifContainer categoria={c} key={c}></GifContainer>
                })
            }
                
            
        </>
    )
}
