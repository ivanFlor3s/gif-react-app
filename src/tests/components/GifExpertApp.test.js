import React from "react";

import { shallow } from "enzyme";
import { GifExpertApp } from "../../gifExpertApp";

describe('Pruebas en componente GifExpertApp', () => {
    
    test('debe de mostrar el componente correctamente', () => {
        
        const wrapper = shallow(<GifExpertApp ></GifExpertApp>)

        expect(wrapper).toMatchSnapshot()
    })

    test('debe de mostrar el componente con sus categorias por props', () => {
        const categorias = ['goku', 'veggeta']
        const wrapper = shallow(<GifExpertApp categoryList={categorias}></GifExpertApp>)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifContainer').length).toBe(categorias.length)
    })
    
})
