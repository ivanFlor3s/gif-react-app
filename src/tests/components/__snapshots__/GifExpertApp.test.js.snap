// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifExpertApp debe de mostrar el componente con sus categorias por props 1`] = `
<Fragment>
  <h1>
    Gif Expert App
  </h1>
  <hr />
  <AddCategory
    addGifHandler={[Function]}
  />
  <h3>
    Categorias
  </h3>
  <GifContainer
    categoria="goku"
    key="goku"
  />
  <GifContainer
    categoria="veggeta"
    key="veggeta"
  />
</Fragment>
`;

exports[`Pruebas en componente GifExpertApp debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h1>
    Gif Expert App
  </h1>
  <hr />
  <AddCategory
    addGifHandler={[Function]}
  />
  <h3>
    Categorias
  </h3>
</Fragment>
`;
